source/coap_connection_handler.c

```diff
@@ -266,7 +266,10 @@ static internal_socket_t *int_socket_create(uint16_t listen_port, bool use_ephem
 
         // Set socket option to receive packet info
         socket_setsockopt(this->socket, SOCKET_IPPROTO_IPV6, SOCKET_IPV6_RECVPKTINFO, &(const bool) {1}, sizeof(bool));
-
+        if (socket_interface_selection != -1) {
+            // Select socket interface if selection requested
+            socket_setsockopt(this->socket, SOCKET_IPPROTO_IPV6, SOCKET_INTERFACE_SELECT, &socket_interface_selection, sizeof(int8_t));
```

Comment:
This is one of the socket options I'd like to deprecate, but the more standard alternative would be too much disruption, so I guess it's staying for now.

Just be aware that it won't religiously obey you - it's pretty much only used for link-local destinations (and maybe multicast?). Wider-scope unicasts will still use the routing table, and ignore this.